scripts/test-migrations.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+function run_tests_for_version() {
+    set -e  # Exit immediately if a command exits with a non-zero status
+    local VERSION=$1
+
+    # Initialize zenml with the appropriate template
+    git clone -b "release/0.43.0" https://github.com/zenml-io/template-starter
+    copier copy template-starter/ test_starter --trust --defaults
+    cd test_starter
+
+    zenml integration install sklearn -y
+    python3 run.py
+    # Add additional CLI tests here
+    zenml version
+
+    # Confirm DB works and is accessible
+    zenml pipeline runs list >> dump.txt && echo "Success listing pipeline runs on $VERSION"
+
+    cd ..
+    rm -rf test_starter template-starter
+}
+
+# List of versions to test
+VERSIONS=("0.39.1" "0.40.0" "0.40.3" "0.41.0" "0.43.0" "0.44.1" "0.44.3" "0.45.2" "0.45.3" "0.45.4" "0.45.5" "0.45.6")
```

Comment:
While this tests that migration works for existing releases, I think the more interesting case would be to consider if the current local files on the PR/branch also work on top of that.